dashboard/app/models/user.rb

```diff
@@ -989,4 +989,8 @@ def can_edit_account?
     return !(sections_as_student.all? {|section| section.login_type == Section::LOGIN_TYPE_PICTURE})
   end
 
+  def section_for_script(script)
+    followeds.collect(&:section).find { |section| section.script_id == script.id }
```

Comment:
I am not convinced that this is the best way to go about doing this, but nor can I clearly see a better way.